payload/vanguard/handler/upload.py

```diff
@@ -0,0 +1,56 @@
+from ..protocol.vanguard import ProgramUploadMsg, ProgramResultMsg
+
+import vanguard.protocol.vanguard
+from ..radio import Radio
+import struct
+import subprocess
+
+
+
+class UploadHandler(object):
+    msg_type = ProgramUploadMsg.TYPE
+    
+    def handle(self, radio, msg):
+        self.radio = radio
+        self.parse_message(msg)
+
+    def parse_message(self, msg):
+        fmt = '!HLBBLHHHHH' #index, chunk, chunk_count, program_name_len, program_data_len
+        unpacked_data = struct.unpack_from(fmt, msg.as_buffer()) 
+        self.index = unpacked_data[5]
+        self.chunk = unpacked_data[6]
+        self.chunk_count = unpacked_data[7]
+        prog_name_length = unpacked_data[8]
+        prog_data_length = unpacked_data[9]
+
+        secondary_fmt = '!HLBBLHHHHH' + str(prog_name_length) + 's' + str(prog_data_length-1) + 'sH'
```

Comment:
would be probably be less error prone to reuse `fmt` instead of having the same exact same format string as above